api/lib/application/tutorial-evaluations/index.js

```diff
@@ -0,0 +1,21 @@
+const tutorialEvaluationsController = require('./tutorial-evaluations-controller');
+
+exports.register = async (server) => {
+  server.route([
+    {
+      method: 'PUT',
```

Comment:
`PUT` me semble pas très approprié comme [verbe](https://developer.mozilla.org/fr/docs/Web/HTTP/M%C3%A9thode) HTTP. Puisqu'on ne remplace pas la ressource mais on la créer si elle n'existe pas. Si elle existe on ne fait rien. Pour de la création il est plutôt recommandé d'utiliser `POST` qui réalise effet de bord dans le système. Une évaluation ne peut jamais être changée ? Ou peut être que c'est prévu au travers d'une autre route de l'API ?

Reply:
En fait, on souhaite tester la fonctionnalité d'évaluation de tutoriel dans un premier temps avant d'y ajouter les méthodes de DELETE et de FIND qui la rendront idempotente à 100%.
Aujourd'hui, on souhaite qu'un click sur un tuto déjà évalué ne rajoute pas de nouvelle entrée en base de donnée (c'est possible quand on rafraîchit le cycle de vie de l'application)

Reply:
`PUT` est parfaitement appropriée pour la création de ressources, notamment quand cette création ne doit avoir lieu que si la ressource n'existe pas encore : https://tools.ietf.org/html/rfc7231#section-4.3.4
> If the target resource does not have a current representation and the
   PUT successfully creates one, then the origin server MUST inform the
   user agent by sending a 201 (Created) response.

Reply:
Oui mais dans l'implémentation actuelle quand on rappelle le endpoint il ne se passe plus rien.

Reply:
Oui, il est idempotent, c'est voulu justement non ?

Reply:
Bah quand tu rappelles un PUT il est sensé remplacer la ressource, non ?

Reply:
Si la représentation de ressource que tu fournis au `PUT` est identique à la représentation existante, "remplacer la ressource" est théoriquement indistingable de "ne rien faire".
Comme actuellement il n'existe qu'une représentation possible de `tutorial-evaluation` (il manque au minimum un `id` d'ailleurs sinon Ember-Data va capoter)… ça colle 🤷 
En revanche je vois que le `PUT` renvoie systématiquement `201`, ça c'est moins propre.
Et personnellement je n'aurais pas séparé `PUT` et `GET` dans des PR séparées, parce que ça rend compliqué l'évaluation du design de l'API 🤔

Reply:
Je vais implémenter le GET